synapse/visibility.py

```diff
@@ -46,6 +46,16 @@
 )
 
 
+async def filter_historical_events(events):
+    filtered_events = [e for e in events if not e.content.get("m.historical", None)]
+
+    # remove the None entries
+    filtered_events = filter(operator.truth, filtered_events)
+
+    # we turn it into a list before returning it.
+    return list(filtered_events)
```

Comment:
This method looks incorrect:
* From the first line `filtered_events` should never contain `None`, so the `filter()` call is unnecessary.
* There's no reason for this to be `async`.
* Please add type hints: `filter_historial_events(Iterable[EventBase]) -> List[EventBase]:`

Reply:
Thanks, I based this on `filter_events_for_client` which doesn't look as necessary for this simple filter :+1: ->

https://github.com/matrix-org/synapse/blob/fb0e14ee9a8cfc9962b2e583cf4837e655d9ca5d/synapse/visibility.py#L78-L80

https://github.com/matrix-org/synapse/blob/fb0e14ee9a8cfc9962b2e583cf4837e655d9ca5d/synapse/visibility.py#L255-L262